.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,53 @@
+{
+	"name": "Eraser",
+	"image": "ghcr.io/cse-labs/k3d:latest",
```

Comment:
what is this image?

Reply:
This image is created by the CSE retail dev crew and has kind installed so individuals using devcontainers/codespaces to develop don't need to go through the installation steps.

Here is the docker file that builds the image: https://github.com/cse-labs/codespaces-images/blob/main/Dockerfile
Here is the link to the package: https://github.com/cse-labs/kubernetes-in-codespaces/pkgs/container/k3d

Reply:
is there a minimal community maintained image we can use here? this image seems to have a lot of tooling built in

Reply:
I worked on using the golang image for codespaces
I was able to install kind, but found that it failed when spinning up cluaster
And then I remembered the talk Bart Robertson gave on setting it up.
Because codespaces is an instance running on a container, and then you're trying to run a cluster via kind.... which is also utilizing a docker container.... we'd be running docker in docker (we named the image dind) which requires additional scripts and settings in order to run
The image with the configs is already made and maintained, so I think it would be preferable to me trying to hobble it together with no intention to maintain it

Reply:
can we install `golang` and `kind` into the container with post create job?

Reply:
We can't but we'd be running docker in docker (we named the image dind) which requires additional scripts and settings in order to run. This image has them already made and maintained, so I think it would be preferable to me trying to hobble it together with no intention to maintain it

Reply:
https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker-in-docker.md
https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker.md

Let me know if you want to jump into a call to discuss further

Reply:
@joheec i don't think this is docker in docker since we are mounting the socket from host https://github.com/Azure/eraser/pull/185/files#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R7